hudi-common/src/main/java/org/apache/hudi/common/config/HoodieReaderConfig.java

```diff
@@ -89,4 +95,18 @@ public class HoodieReaderConfig extends HoodieConfig {
       "hoodie.write.record.merge.custom.implementation.classes";
   public static final String RECORD_MERGE_IMPL_CLASSES_DEPRECATED_WRITE_CONFIG_KEY =
       "hoodie.datasource.write.record.merger.impls";
+
+  public static boolean isFileGroupReaderEnabled(HoodieTableVersion tableVersion, HoodieConfig config) {
+    return tableVersion.greaterThanOrEquals(HoodieTableVersion.EIGHT) && config.getBooleanOrDefault(HoodieReaderConfig.FILE_GROUP_READER_ENABLED);
```

Comment:
We should not differentiate the reader logic at this layer.  If there are gaps regarding the file group reader reading the base and log files written in table version 6, let's fix that within the file group reader.  Otherwise, we'll never deprecate the legacy log scanner and reader path.

Reply:
https://github.com/apache/hudi/pull/12935